tools/single_use/SaveUserIDToInstrumentData.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * This script is intended for a one-time use only to restore the value of the
+ * `UserID` column of instrument tables and the `UserID` key of the instrument
+ * JSON `Data` in the flag table.
+ * 
+ * This tool queries data from the `history` table and meaningfully imports
+ * data from its `userID` column back into the instrument and `flag` tables.
+ *
+ * Example use:
+ * $ php SaveUserIDToInstrumentData.php [confirm]
+
+ * PHP Version 7
+ *
+ * @category Tools
+ * @package  Loris
+ * @author   Zaliqa Rosli <zaliqa.rosli@mcin.ca>
+ * @licence  LORIS License
+ * @link     https://www.github.com/aces/Loris
+ */
+require_once __DIR__ . '/../generic_includes.php';
+
+$confirm = false;
+if (isset($argv[1]) && $argv[1] === 'confirm') {
+    $confirm = true;
+}
+
+echo "\n\nQuerying data...\n";
+
+$db = \NDB_Factory::singleton()->database();
+
+$result_count = 1;
+foreach(\Utility::getAllInstruments() as $table => $name) {
```

Comment:
@zaliqarosli This line is incorrect. the `$table` is actually the `$testname` and the `$name` is actually the `$Full_name`

in order to get the `table` you need to instanciate the instrument and request the `table` variable !

A good example to follow is the fix_candidate_age tool in this case. Including the `tableExists()` and `columnExists()` checks in the script to make sure not to get an SQL error.
https://github.com/aces/Loris/blob/23.0-release/tools/data_integrity/fix_candidate_age.php#L46



You might also wanna add a check for JSONData instruments since they would not have a table in the database
https://github.com/aces/Loris/blob/23.0-release/tools/CouchDB_Import_Instruments.php#L139

Reply:
After Testing i also realized the same comment ID is being returned multiple times indicating that something might be wrong with the query.